taichi/rhi/opengl/opengl_api.cpp

```diff
@@ -27,8 +27,12 @@ static bool kUseGles = false;
 static std::optional<bool> supported;  // std::nullopt
 static bool context_with_glfw = false;
 void *kGetOpenglProcAddr;
+bool kSetGles = false;
```

Comment:
I think GLES "override" might be a better name.

In addition to that, setting a global variable directly doesn't feel too good, might have weird ABI or linking issues. I would prefer this kind of solution with anonymous namespace & std::optional & setter
```c++
namespace {
static std::optional<bool> use_gles_override;
};

static void set_gles_override(bool value) {
  use_gles_override = value;
}

static void unset_gles_override(bool value) {
  use_gles_override = std::nullopt;
}
```

This way later down the code this flag can force GLES both ways

Reply:
And the headers now only need to contain declarations of these two functions

Reply:
Thank you for the feedback. I've updated the pull request with the changes you suggested. If you have any other feedback or suggestions, that would be appreciated.